### PR TITLE
dynamodb revisionGuardStore implementation

### PR DESCRIPTION
**Pull request by emmkong**

### CODE REVIEW

**Comment by nanov** [comment]
I pretty sure "aws-sdk" should be in devDependencies section, as well as [this line](https://github.com/orchestrated-io/node-cqrs-eventdenormalizer/blob/42bf8f5b77fe311107cc7876cdf43ed4308e652c/lib/revisionGuardStore/databases/dynamodb.js#L10) should be :
```javascript
aws = Store.use('aws-sdk');
```

in order to not make it hard dependency.

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
   },
   "dependencies": {
     "async": "1.5.2",
+    "aws-sdk": "2.94.0",
```
Comment (adrai) [inline package.json]: move this to devDependencies, please

File: lib/revisionGuardStore/databases/dynamodb.js [hunk 2]
```diff
@@ -0,0 +1,325 @@
+'use strict';
+
+var util = require('util'),
+  Store = require('../base'),
+  debug = require('debug')('denormalizer:revisionGuardStore:dynamodb'),
+  ConcurrencyError = require('../../errors/concurrencyError'),
+  async = require('async'),
+  _ = require('lodash'),
+  uuid = require('uuid').v4,
+  aws = require('aws-sdk');
```
Comment (adrai) [inline lib/revisionGuardStore/databases/dynamodb.js]: please write it that way: `aws = Store.use('aws-sdk');`

**Comment by adrai** [comment]
@emmkong seen the comments?

**Comment by emmkong** [comment]
@adrai @nanov , thanks for pointing this out. I have updated the files as per suggested. Please have a look. Thanks

**Pull request merged**